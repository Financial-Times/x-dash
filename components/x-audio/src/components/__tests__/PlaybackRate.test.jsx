const { h } = require('@financial-times/x-engine');
const { shallow } = require('@financial-times/x-test-utils/enzyme');
const { PlaybackRate } = require('../PlaybackRate');

jest.mock('../playback-rates', () => {
	return { getNextPlaybackRate: () => 1.5 };
});

jest.mock('../classnames-helper', () => {
	return (...classes) => classes.join(' ');
});


describe('PlaybackRate', () => {

	const props = { onClick: jest.fn() };
	const subject = shallow(<PlaybackRate {...props}/>);

	test('should have necessary elements', () => {
		expect(subject).toMatchSnapshot();
	});

	test('should call onClick function with next playbackRate when it is clicked', () => {
		subject.simulate('click');
		expect(subject).toMatchSnapshot();
		expect(props.onClick).toHaveBeenCalledWith(1.5);
	});
});
