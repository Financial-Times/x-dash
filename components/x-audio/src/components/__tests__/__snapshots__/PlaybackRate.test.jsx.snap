// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaybackRate should call onClick function with next playbackRate when it is clicked 1`] = `
<button
  aria-label="Change playback speed to 1.5 times"
  className="audio-player__control-speed"
  onClick={[Function]}
>
  1
  x
</button>
`;

exports[`PlaybackRate should have necessary elements 1`] = `
<button
  aria-label="Change playback speed to 1.5 times"
  className="audio-player__control-speed"
  onClick={[Function]}
>
  1
  x
</button>
`;
